File: src/main/java/com/loltft/rudefriend/entity/SaveFile.java
package com.loltft.rudefriend.entity;

import java.time.LocalDateTime;

import org.springframework.data.annotation.CreatedBy;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import com.fasterxml.jackson.annotation.JsonFormat;

import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EntityListeners;
import jakarta.persistence.FetchType;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "save_file")
@Getter
@Builder(toBuilder = true)
@AllArgsConstructor
@NoArgsConstructor
@EntityListeners(AuditingEntityListener.class)
public class SaveFile {

  @Id
  @Schema(description = "extension 포함 된 저장된 파일명")
  private String fileUuid;

  @Column
  @Schema(description = "원본 파일명")
  private String originalFileName;

  @CreatedDate
  @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
  @Schema(description = "업로드 일시")
  private LocalDateTime uploadDateTime;

  @CreatedBy
  @Schema(description = "업로더")
  @ManyToOne(optional = false, fetch = FetchType.LAZY)
  private Member uploader;
}

File: src/main/java/com/loltft/rudefriend/entity/SaveFile.kt
package com.loltft.rudefriend.entity

import com.fasterxml.jackson.annotation.JsonFormat
import io.swagger.v3.oas.annotations.media.Schema
import jakarta.persistence.*
import org.springframework.data.annotation.CreatedBy
import org.springframework.data.annotation.CreatedDate
import org.springframework.data.jpa.domain.support.AuditingEntityListener
import java.time.LocalDateTime
import java.util.UUID

@Entity
@Table(name = "save_file")
@EntityListeners(AuditingEntityListener::class)
class SaveFile(
    @Id
    @Schema(description = "extension 포함 된 저장된 파일명")
    var fileUuid: UUID,

    @Column
    @Schema(description = "원본 파일명")
    var originalFileName: String,

    @CreatedDate
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    @Schema(description = "업로드 일시")
    var uploadDateTime: LocalDateTime,

    @Column
    @Schema(description = "S3 버킷 내의 폴더명")
    var dirName: String,

    @Column
    @Schema(description = "S3에 업로드 된 파일의 전체 URL")
    var fullUrl: String,

    @Column
    @Schema(description = "저장한 게시글 PK")
    var boardId: UUID
)
